src/preprocessing/filtering.jl

```diff
@@ -1,88 +1,145 @@
-"""
-    highpass_filter(signals::Array{T}, cutOff::Number, fs::Number, order::Int) where {T<:AbstractFloat}
 
-High pass filter applied in forward and reverse direction
+	
+	
+	
+	function filter(obj::NeuroimagingMeasurement,responsetype::FilterType; kwargs...)
+		# Afaik there is a discussion going on, whether such error messages should be provided or not. I think I like that this is here, because you need less julia knowledge to understand what is wrong
+		error("Currently no filter defaults implemented for $(typeof(obj)), please see the documentation how to define your own filter or raise an issue on GitHub.")
+	end
+	
+	# Modality independent filtering
+	function filter(
+			obj::NeuroimagingMeasurement,
+			responsetype::FilterType,
+			designmethod;kwargs...
+			)		
+		
+		fobj = digitalfilter(responsetype,designmethod)
+        obj2 = deepcopy(obj) # TODO: Not sure if this is the proper way to make sure data is not overwritten inplace
+		obj2.data = filter(obj2.data,fobj;kwargs...)
+		return obj2
+	end
+		
+	function filter(s,fobj; filtfilt=false)

```

Comment:
No defaults at the low level please. 
```suggestion
	function filter(s, fobj, filtfilt)
```